### ISSUE TITLE
Introduce "Go Back" and "Go Forward" to navigate between recent edits

### ISSUE THREAD

**Issue by johnbillion** [Microsoft/vscode#42617]
- VSCode Version: Code 1.19.3 (7c4205b5c6e52a53b81c69d2b2dc8a627abaa0ba, 2018-01-25T10:25:01.248Z)
- OS Version: Darwin x64 16.4.0

The existing `Go Back` and `Go Forward` commands navigate between recent points where the cursor has been. It would be great if there was an addition to these commands to navigate only between points where edits have been made (ie. ignoring points where the cursor has been but not made an edit).

I often find that I make an edit, then perform several pieces of code navigation, and then want to quickly jump back to my previous editing location. If I'm in the same file I usually do a quick undo and redo to move the cursor back to the previous editing position.

**Comment by MindfulMe** [comment]
So does it implemented?😕

**Comment by MindfulMe** [comment]
Yes, it is already in VS code.
![screen shot 2018-09-03 at 12 08 14 pm](https://user-images.githubusercontent.com/26275425/44977911-0f24c280-af72-11e8-9c8d-150cfec7c696.png)

**Comment by cipri-tom** [comment]
I believe `Back` and `Forward` should have an option to stay in the same file. In other words, if I just switch from a different tab and I press back it should take me where I previously was in that file, not to the tab that I just switched from.

**Comment by Otiel** [comment]
@cipri-tom Please vote on #36634

**Comment by houh60** [comment]
Please correct me if I'm wrong. I think what we're talking about is to go back to the previous position or go forward to the next position we have been, right? 

If that's the case, we use "alt + left arrow" to go back and "alt + right arrow" to go forward.

**Comment by MindfulMe** [comment]
Okay

**Comment by rlaferla** [comment]
This is already in VSCode as menu items BUT what we need are two big buttons on the UI to go "back" and "forward".  It would really improve productivity.  Please make it happen.

**Comment by johnbillion** [comment]
Yes the back now exists (as `Go to Last Edit Location`) and it's indeed very useful, but does the corresponding forward command exist?

**Comment by kchenzhi** [comment]
Hi, is there a way to go forward/back when I reading the source code in maven's dependency jar?
when I go back, the editor jump to a common source file which I latest edited in my project .

**Comment by bpasero** [comment]
**Verification**
* edit at multiple locations in different text editors across different editor groups
* use the new commands to navigate in edit locations and verify they work as advertised

**Note:** "Go Previous in Edit Locations" acts like a toggle for going back and forward, i.e. jumping between the current and the previous location.
